tslib/src/functions-throttled/sync-results-to-sql/sync-results-data.service.spec.ts

```diff
@@ -0,0 +1,102 @@
+import * as R from 'ramda'
+import { mockCompletionCheckMessage } from './mocks/completed-check.message'
+import { SyncResultsDataService } from './sync-results.data.service'
+import { ISqlService } from '../../sql/sql.service'
+import { DBEventType } from './models'
+
+const mockQuestionData = new Map()
+mockQuestionData.set('1x1', { id: 1, factor1: 1, factor2: 2, isWarmup: false, code: 'Q001' })
+const validatedCheck = mockCompletionCheckMessage.validatedCheck
+let mockSqlService: ISqlService
+
+describe('SyncResultsDataService', () => {
+  let sut: SyncResultsDataService
+  let date: Date
+
+  beforeEach(() => {
+    mockSqlService = {
+      query: jest.fn(),
+      modify: jest.fn(),
+      modifyWithTransaction: jest.fn()
+    }
+    sut = new SyncResultsDataService(mockSqlService)
+    date = new Date()
+  })
+
+  it('calls `createNewEvent` if a new event is seen in the audit log when preparing events', async () => {
+    // setup
+    const eventTypes = new Map<string, DBEventType>()
+    eventTypes.set('TestEvent1', { id: 1, eventType: 'TestEvent1', eventDescription: '' })
+    eventTypes.set('TestEvent2', { id: 1, eventType: 'TestEvent2', eventDescription: '' })
+
+    jest.spyOn(sut as any, 'createNewEventType').mockImplementation(jest.fn())
```

Comment:
seems odd to need to cast down to `any`. was this causing an issue?

Reply:
Spied methods in typescript can be typed as two types: the original object and the jest mock.  This is one method to convert a private method to jest spy.

If it was typed as a SyncResultDataService typescript would not allow it due to it being private.   Often private methods would not be covered when unit testing public interfaces, but this one has a side effect of inserting something into the db.

If there is a better way please let me know.

Reply:
My understanding is that reflection in typescript is pretty poor.  Another way would be to subclass the test class and make the variable public.

It seems easier to just lean on the fact that this is transpiled to javascript.

Reply:
i see why you are using this now, because it allows you to peek in and inspect the internal variables using reflection.

there are 'different' ways to do it, but that would involve pretty much rewriting what you already have, which isn't really viable for us, as we are coming from a javascript implementation to typescript, rather than writing this from the ground up.